tests/test_ilastik/test_applets/base/testSerializer.py

```diff
@@ -535,6 +536,56 @@ def iter_dataset_names(name):
         shutil.rmtree(tmp_dir)
 
 
+@pytest.fixture
+def opLabelArray():
+    raw_data = numpy.zeros((256, 256, 256, 1), dtype=numpy.uint32)
+    opLabelArrays = OperatorWrapper(OpCompressedUserLabelArray, graph=Graph())
+    opLabelArrays.Input.resize(1)
+    opLabelArrays.Input[0].meta.axistags = vigra.AxisTags("zyxc")
+    opLabelArrays.Input[0].setValue(raw_data)
+    opLabelArrays.shape.setValue(raw_data.shape)
+    opLabelArrays.eraser.setValue(255)
+    opLabelArrays.deleteLabel.setValue(-1)
+    opLabelArrays.blockShape.setValue((64, 64, 64, 1))
+    return opLabelArrays
+
+
+def testCompression(tmpdir, opLabelArray):
+    def has_uncompressed_dataset(obj_name, obj):
+        check = any(pref in obj_name for pref in ["block", "data"])
+        if check and isinstance(obj, h5py.Dataset) and not obj.compression:
+            return obj_name
+
+    h5_filepath_no_compression = tmpdir / "serial_blockslot_no-compression.h5"
+    h5_filepath_compressed = tmpdir / "serial_blockslot_compressed.h5"
+
+    # Create an operator and a serializer to write the data.
+    slotSerializer_no_compression = SerialBlockSlot(opLabelArray.Output, opLabelArray.Input, opLabelArray.nonzeroBlocks)
+    slotSerializer_compressed = SerialBlockSlot(
+        opLabelArray.Output, opLabelArray.Input, opLabelArray.nonzeroBlocks, compression_level=1
+    )
+    # Give it some data.
+    opLabelArray.Input[0][0:1, 0:1, 0:1, 0:1] = 1 * numpy.ones((1, 1, 1, 1), dtype=numpy.uint8)
+
+    with h5py.File(h5_filepath_no_compression, "w") as f:
+        label_group = f.create_group("label_data")
+        slotSerializer_no_compression.serialize(label_group)
+        uncompressed = label_group.visititems(has_uncompressed_dataset)
+        assert uncompressed, f"Expected uncompressed dataset"
+
+    assert h5_filepath_no_compression.exists()
+
+    with h5py.File(h5_filepath_compressed, "w") as f:
+        label_group = f.create_group("label_data")
+        slotSerializer_compressed.serialize(label_group)
+        uncompressed = label_group.visititems(has_uncompressed_dataset)
+        assert uncompressed is None, f"Found unexpected uncompressed dataset {uncompressed}"
+
+    assert h5_filepath_compressed.exists()
+
+    assert h5_filepath_compressed.stat().size < h5_filepath_no_compression.stat().size
```

Comment:
Do we still need this? Strictly speaking, compressed data could be larger (the maximum entropy plus compression bookkeeping case).

Reply:
thx, no we don't!